labgrid/driver/bareboxdriver.py

```diff
@@ -33,9 +35,11 @@ class BareboxDriver(CommandMixin, Driver, CommandProtocol, LinuxBootProtocol):
     prompt = attr.ib(default="", validator=attr.validators.instance_of(str))
     autoboot = attr.ib(default="stop autoboot", validator=attr.validators.instance_of(str))
     interrupt = attr.ib(default="\n", validator=attr.validators.instance_of(str))
+    boot_expression = attr.ib(default=r"[\n]barebox 20\d+", validator=attr.validators.instance_of(str))
     bootstring = attr.ib(default=r"Linux version \d", validator=attr.validators.instance_of(str))
     password = attr.ib(default="", validator=attr.validators.instance_of(str))
     login_timeout = attr.ib(default=60, validator=attr.validators.instance_of(int))
+    boot_detected = attr.ib(default=False, validator=attr.validators.instance_of(int))
```

Comment:
This should be `instance_of(bool)`. We also probably want to set this to `setattr=None`, so users can't change this attribute.

Reply:
I fixed the type.
With attr I also found no way to protect it. Not sure if it worth `_boot_detected` and `get_boot_detected` like done with `_status`